### DISCUSSION TITLE
Get mangled function name

### DISCUSSION THREAD

**Discussion by mixern6** [NationalSecurityAgency/ghidra#7110] [Q&A] [answered]
Hello, I'm using FlatAPI in python and I'm trying to get the original mangled name (like `_Z8glob_fn9cilxjmfdebPvPS_ `) of a function. Here is what I tried with no success. All of these return the demangled name

```
          json_data["name"] = function.getName()

          function_symbol = flat_api.getSymbolAt(function_entry_point)
          json_data["functionsymbolname"] = function_symbol.getName()

          symbol = function.getSymbol()
          json_data["symbolname"] = symbol.getName()

          signature = function.getSignature()
          json_data["signaturename"] = signature.getName()
```

**Comment by ryanmkurtz** [comment, marked as answer]
I'm not sure if this is optimal, but you can do `for entry in currentProgram.getSymbolTable().getLabelHistory(addr): ...` and try to get it from there.  `entry` is of type `LabelHistory` which you can look up.

**Comment by dragonmacher** [comment]
Each mangled symbol should still be at the address of the function, along with the demangled symbol.   It looks like there is no method on `FlatProgramAPI` to get them, but you can from the `SymbolManager`:

```java
SymbolManager symbolManager = currentProgram.getSymbolManager();
Symbol[] symbols = symbolManager.getSymbols(function_entry_point);
```
